src/EnergyPlus/SystemAvailabilityManager.cc

```diff
@@ -1380,33 +1341,33 @@ namespace SystemAvailabilityManager {
                 auto const &objectFields = instance.value();
                 auto const &thisObjectName = UtilityRoutines::MakeUPPERCase(instance.key());
                 ip->markObjectAsUsed(cCurrentModuleObject, instance.key());
-                state.dataSystemAvailabilityManager->SysAvailMgrListData(Item).Name = thisObjectName;
+                auto &mgrList = state.dataSystemAvailabilityManager->ListData(Item);
+                mgrList.Name = thisObjectName;
 
                 auto extensibles = objectFields.find("managers");
                 auto const &extensionSchemaProps = objectSchemaProps["managers"]["items"]["properties"];
                 if (extensibles != objectFields.end()) {
                     auto extensiblesArray = extensibles.value();
                     int numExtensibles = extensiblesArray.size();
-                    state.dataSystemAvailabilityManager->SysAvailMgrListData(Item).NumItems = numExtensibles;
-                    state.dataSystemAvailabilityManager->SysAvailMgrListData(Item).AvailManagerName.allocate(numExtensibles);
-                    state.dataSystemAvailabilityManager->SysAvailMgrListData(Item).cAvailManagerType.allocate(numExtensibles);
-                    state.dataSystemAvailabilityManager->SysAvailMgrListData(Item).AvailManagerType.allocate(numExtensibles);
+                    mgrList.NumItems = numExtensibles;
+                    mgrList.AvailManagerName.allocate(numExtensibles);
+                    mgrList.AvailManagerType.allocate(numExtensibles);
                     for (int extItem = 1; extItem <= numExtensibles; ++extItem) {
-                        state.dataSystemAvailabilityManager->SysAvailMgrListData(Item).AvailManagerName = "";
-                        state.dataSystemAvailabilityManager->SysAvailMgrListData(Item).cAvailManagerType = "";
-                        state.dataSystemAvailabilityManager->SysAvailMgrListData(Item).AvailManagerType = 0;
+                        mgrList.AvailManagerName = "";
+                        mgrList.AvailManagerType = DataPlant::SystemAvailabilityType::Invalid;
                     }
 
                     int listItem = 0;
-                    for (auto extensibleInstance : extensiblesArray) {
+                    for (const auto& extensibleInstance : extensiblesArray) {
```

Comment:
Should I revert this? I made this change based on a [ClangTidy suggestion](https://clang.llvm.org/extra/clang-tidy/checks/performance-for-range-copy.html), but I was not completely sure if extensibleInstance is not a scalar.

Reply:
If only there was an alternative to `auto` that would tell you what the type of a variable was... 😡

Reply:
`extensibleInstance` is of type `nlohmann::basic_json`. I'm pretty new to that one...